docs/modeling/max-length.rst

```diff
@@ -0,0 +1,39 @@
+.. include:: /rc1-notice.txt
+
+Maximum Length
+==============
+
+Configuring a maximum length provides a hint to the data store about the appropriate data type to use for a given property. Maximum length only applies to array data types, such as ``string`` and ``byte[]``.
+
+.. note::
+    Entity Framework does not do any validation of maximum length before passing data to the provider. It is up to the provider or data store to validate if appropriate. For example, when targeting SQL Server, exceeding the maximum length will result in an exception as the data type of the underlying column will not allow excess data to be stored.
```

Comment:
And for comparison, when targeting SQLite, exceeding the maxlength does not throw an error because SQLite does not have max length contraints.